### ISSUE TITLE
Any idea of using CLI to delete the project?

### ISSUE THREAD

**Issue by TaroballzChen** [railwayapp/cli#157]
I would like to use the `railway` command to delete the specific project, any idea to do this?

**Comment by JakeCooper** [comment]
Definitely possible. If you wanna add it to the CLI, it’s just deleteProject(Id) in our GraphQL API

**Comment by TaroballzChen** [comment]
I add `delete.go` document in `cmd/` folder and the content as follow:
```go
package cmd

import (
	"context"
	"fmt"

	"github.com/railwayapp/cli/entity"
	"github.com/railwayapp/cli/ui"
	"github.com/railwayapp/cli/uuid"
)

func(h *Handler)Delete(ctx context.Context, req *entity.CommandRequest)error{
	if len(req.Args) > 0 {
		// projectID provided as argument
		arg := req.Args[0]

		if (uuid.IsValidUUID(arg)) {
			project, err := h.ctrl.GetProject(ctx, arg)
			if (err != nil) {
				return err
			}

			return h.ctrl.DeleteProject(ctx, project.Id)
		}

		project, err := h.ctrl.GetProjectByName(ctx, arg)
		if (err != nil) {
			return err
		}

		return h.ctrl.DeleteProject(ctx, project.Id)
	}

	h.Unlink(ctx, req)

	isLoggedIn, err := h.ctrl.IsLoggedIn(ctx)
	if err != nil {
		return err
	}

	if isLoggedIn {
		return h.deleteFromAccount(ctx, req)
	} else {
		return h.deleteFromID(ctx, req)
	}

}

func (h *Handler) deleteFromAccount(ctx context.Context, req *entity.CommandRequest) error {
	projects, err := h.ctrl.GetProjects(ctx)
	if err != nil {
		return err
	}

	if len(projects) == 0 {
		fmt.Printf("No Projects could be deleted.")
		return nil
	}

	project, err := ui.PromptProjects(projects)
	if err != nil {
		return err
	}

	return h.ctrl.DeleteProject(ctx, project.Id)
}

func (h *Handler) deleteFromID(ctx context.Context, req *entity.CommandRequest) error {
	projectID, err := ui.PromptText("Enter your project id")
	if err != nil {
		return err
	}

	project, err := h.ctrl.GetProject(ctx, projectID)
	if (err != nil) {
		return err
	}

	return h.ctrl.DeleteProject(ctx, project.Id)
}
```
append `DeleteProject` function in `controller/project.go`

```go
func (c *Controller) DeleteProject(ctx context.Context, projectID string) error {
	return c.gtwy.DeleteProject(ctx, projectID)
}
```

and append `addRootCmd` execution about deleting project operation in `init` function which in project's `main.go` as follow
```go
	addRootCmd(&cobra.Command{
		Use:   "delete",
		Short: "Delete Project, may specify projectId as an argument",
		RunE:  contextualize(handler.Delete, handler.Panic),
	})
```

the result as follow
![image](https://user-images.githubusercontent.com/27862593/135391921-97633e8a-dbe0-4580-a5c1-1f1a75290d85.png)

the delete operation seems to unsuccessful, any idea?

**Comment by ndneighbor** [comment]
Hey! @TaroballzChen I will be reviewing this once I wake up and working through on why it might not be working. The good news is that the CLI is using all of the same gQL mutations as our front-end so I can check that out and try it against your changes.

**Comment by TaroballzChen** [comment]
Hi! @ndneighbor I'm grateful to hear that.  If any good news during checking, plz guide me how to do that!

**Comment by ndneighbor** [comment]
Alright, got some news for you- I looked into `deleteProject` in the controller of the project and it matches the front-end call. I am going to review with the team to see if there is something else that might cause the 400. 

(I also noticed that if you try to delete a project with an id that doesn't exist it errors out too, so I want to clean up that flow for you as well.) 

Thank you for your patience.
